projects/cds/src/merchandising/model/strategy-result.model.ts

```diff
@@ -7,7 +7,7 @@ export interface MerchandisingProduct {
   thumbNailImage?: string;
   mainImage?: string;
   price?: number;
-  metadata?: Map<string, string>;
+  metadata?: { [metadataAttributeName: string]: string };
```

Comment:
Was there a serialization problem and you had to change the type?
Just curious.

Reply:
Yea, the metadata "map" comes back as a JSON object which we assumed TypeScript would deserialise into a TypeScript `Map`, but when we came to use it to populate the attributes in the div we saw that it was still a plain JSON object, so we got runtime exceptions that methods of `Map` weren't available. 

So we've made it an object here and then in normalizer we make a `Map` from the object

Reply:
Ok, that's what I thought. `Map` is not serializable, so that's probably the reason